### PR TITLE
Cleanup deprecated usage and warnings

### PR DESCRIPTION
**Pull request by levithomason**
This PR cleans up deprecated component access in tests and fixes other warnings thrown during testing.

### CODE REVIEW

File: src/views/Statistic/Statistic.js [hunk 1]
```diff
@@ -11,7 +10,7 @@ import StatisticValue from './StatisticValue'
 
 export default class Statistic extends Component {
   static propTypes = {
-    children: customPropTypes.ofComponentTypes(['Statistics', 'Label', 'Value']),
+    children: PropTypes.node,
```
Comment (levithomason) [inline src/views/Statistic/Statistic.js:13]: A Statistic can render more than just statistic parts.  It is not the documented use but it handles it OK.

File: test/specs/collections/Table/Table-test.js [hunk 2]
```diff
@@ -243,7 +243,7 @@ describe('Table', () => {
     it('is not auto applied by default', () => {
       render(<Table />)
         .findClass('sd-table')
-        .props.className
+        .getAttribute('class')
```
Comment (levithomason) [inline test/specs/collections/Table/Table-test.js:246]: Do not access props of a DOM node, in this case `<table />`.

File: test/specs/modules/Modal/Modal-test.js [hunk 3]
```diff
@@ -23,17 +23,17 @@ describe('Modal', () => {
...
   describe('ModalContent', () => {
     it('should render children', () => {
-      render(<ModalContent>Hello</ModalContent>).assertText('Hello')
+      render(<Modal.Content>Hello</Modal.Content>).assertText('Hello')
     })
   })
   describe('ModalFooter', () => {
     it('should render children', () => {
-      render(<ModalFooter>Hey You</ModalFooter>).assertText('Hey You')
+      render(<Modal.Footer>Hey You</Modal.Footer>).assertText('Hey You')
```
Comment (levithomason) [inline test/specs/modules/Modal/Modal-test.js:36]: Use sub component access, the top level components are deprecated.

**Comment by dvdzkwsk** [comment]
:ghost:

**Comment by kyleturco** [comment]
🤖

**Pull request merged**